src/components/Logo/QueryTokenLogo.tsx

```diff
@@ -1,32 +1,21 @@
-import { NATIVE_CHAIN_ID } from 'constants/tokens'
-import { TokenStandard } from 'graphql/data/__generated__/types-and-hooks'
+import { ChainId } from '@uniswap/sdk-core'
+import { PortfolioLogo } from 'components/AccountDrawer/MiniPortfolio/PortfolioLogo'
 import { SearchToken } from 'graphql/data/SearchTokens'
 import { TokenQueryData } from 'graphql/data/Token'
 import { TopToken } from 'graphql/data/TopTokens'
-import { supportedChainIdFromGQLChain } from 'graphql/data/util'
+import { gqlToCurrency, supportedChainIdFromGQLChain } from 'graphql/data/util'
+import { useMemo } from 'react'
 
-import AssetLogo, { AssetLogoBaseProps } from './AssetLogo'
+import { AssetLogoBaseProps } from './AssetLogo'
 
 export default function QueryTokenLogo(
   props: AssetLogoBaseProps & {
     token?: TopToken | TokenQueryData | SearchToken
   }
 ) {
-  const chainId = props.token?.chain ? supportedChainIdFromGQLChain(props.token?.chain) : undefined
+  const chainId =
+    (props.token?.chain ? supportedChainIdFromGQLChain(props.token?.chain) : ChainId.MAINNET) ?? ChainId.MAINNET
+  const currency = props.token ? gqlToCurrency(props.token) : undefined
 
-  return (
-    <AssetLogo
-      isNative={
-        // TODO(cartcrom): simplify this check after backend fixes token standard on assetActivities tokens
-        !props.token?.address ||
-        props.token?.standard === TokenStandard.Native ||
-        props.token?.address === NATIVE_CHAIN_ID
-      }
-      chainId={chainId}
-      address={props.token?.address}
-      symbol={props.token?.symbol}
-      backupImg={props.token?.project?.logoUrl}
-      {...props}
-    />
-  )
+  return <PortfolioLogo currencies={useMemo(() => [currency], [currency])} chainId={chainId} {...props} />
```

Comment:
another idea is instead of declaring `currency` as the constant above, make it 

```
const currencies = useMemo(() => {
   return props.token ? [gqlToCurrency(props.token)] : undefined
}, [props.token])

return <PortfolioLogo currencies={currencies} chainId={chainId} {...props} />
```